marahgh/assignment1/Q1.cpp

```diff
@@ -0,0 +1,171 @@
+
+
+#include <iostream>
+#include<vector>
+using namespace std;
+
+/*anagram words */
+//My assumption:the words alphabet is english letters.    
+
+
+//---------------------------------------------------------------------------------------------------------
+/*A function that determines if two strings are anagrams ,
+	able to handle both case sensitive and case insensitive anagrams,
+	according to value of caseSensetive flag variable */
+bool areAnagrams(string s1, string s2, bool caseSensetive){
+    const int LEN= 'z' -'a' +1;
+
+    if(s1.size()!= s2.size()){
+        return false;
+    }
+    vector<int> histogram(LEN*2,0);

```

Comment:
Note that the code will be much simpler if you actually don't have the restriction on only english letters :)